### ISSUE TITLE
Error: Cannot find module '../src/cli'

### ISSUE THREAD

**Issue by ShamariFeaster** [firebase/blaze_compiler#59]
I installed globally as specified. blaze is looking for '../src/cli.js' as if it is installed along with the node common modules. Maybe I have something misconfigured but your help would be appreciated.

**Comment by tomlarkworthy** [comment]
So the typescript compiler should run and generate that file. When I install I see the following output.
```
larkworthy-macbookpro:dev larkworthy$ npm install -g blaze_compiler
 
> blaze_compiler@0.0.35 prepublish /Users/larkworthy/dev/blaze_compiler
> grunt compile

Running "typescript:base" (typescript) task
34 files created. js: 17 files, map: 17 files, declaration: 0 files (2376ms)

Done, without errors.
/Users/larkworthy/.nvm/versions/node/v0.12.7/bin/blaze -> /Users/larkworthy/.nvm/versions/node/v0.12.7/lib/node_modules/blaze_compiler/bin/blaze.js
blaze_compiler@0.0.35 /Users/larkworthy/.nvm/versions/node/v0.12.7/lib/node_modules/blaze_compiler
├── tv4@1.1.12
├── xregexp@2.0.0
├── source-processor@0.0.6
├── falafel@0.3.1
├── optimist@0.6.1 (wordwrap@0.0.3, minimist@0.0.10)
└── source-map-support@0.3.2 (source-map@0.1.32)


larkworthy-macbookpro:dev larkworthy$ blaze
run:-
blaze <file>
to generate a rules.json in the current directory, ready for upload to Firebase

Options:
  -v  enable debug output
```

**Comment by tomlarkworthy** [comment]
can you paste me your output?

**Comment by ShamariFeaster** [comment]
It appears as though the TS compiler is not running. 

$ npm install -g blaze_compiler
npm http GET https://registry.npmjs.org/blaze_compiler
npm http 200 https://registry.npmjs.org/blaze_compiler
npm http GET https://registry.npmjs.org/falafel/0.3.1
npm http GET https://registry.npmjs.org/source-processor/0.0.6
npm http GET https://registry.npmjs.org/tv4/1.1.12
npm http GET https://registry.npmjs.org/xregexp/2.0.0
npm http GET https://registry.npmjs.org/optimist
npm http GET https://registry.npmjs.org/source-map-support/0.3.2
npm http 304 https://registry.npmjs.org/optimist
npm http 304 https://registry.npmjs.org/falafel/0.3.1
npm http 304 https://registry.npmjs.org/source-map-support/0.3.2
npm http 304 https://registry.npmjs.org/xregexp/2.0.0
npm http 304 https://registry.npmjs.org/tv4/1.1.12
npm http 304 https://registry.npmjs.org/source-processor/0.0.6
npm http GET https://registry.npmjs.org/wordwrap
npm http GET https://registry.npmjs.org/minimist
npm http GET https://registry.npmjs.org/source-map/0.1.32
npm http 304 https://registry.npmjs.org/wordwrap
npm http 304 https://registry.npmjs.org/minimist
npm http 304 https://registry.npmjs.org/source-map/0.1.32
npm http GET https://registry.npmjs.org/amdefine
npm http 304 https://registry.npmjs.org/amdefine
C:\Users\wzhjtn\AppData\Roaming\npm\blaze -> C:\Users\wzhjtn\AppData\Roaming\npm
\node_modules\blaze_compiler\bin\blaze.js
blaze_compiler@0.0.35 C:\Users\wzhjtn\AppData\Roaming\npm\node_modules\blaze_com
piler
├── tv4@1.1.12
├── xregexp@2.0.0
├── source-processor@0.0.6
├── falafel@0.3.1
├── optimist@0.6.1 (wordwrap@0.0.3, minimist@0.0.10)
└── source-map-support@0.3.2 (source-map@0.1.32)

**Comment by tomlarkworthy** [comment]
Oh actually that makes sense it should be done before upload to npm. I am trying it on another computer and I get similar output to yours. I think my first attempt was being clever as it found where the dev install was. I'll figure it out in a moment

**Comment by tomlarkworthy** [comment]
Ok its installing ok. Just noticed you're Windows ... I have never tested that. Maybe you can help me figure it out. If you go to the npm install directory "C:\Users\wzhjtn\AppData\Roaming\npm\" and find the blaze install directory. One you find the root of the installation you should be able to "node bin/blaze.js" and re-trigger the error. Its probably the file path seperator. Try fiddling with the source code in bin/blaze.js and see what fixes it, then I can try to roll that out after testing on linux

**Comment by ShamariFeaster** [comment]
Node require always uses forward slashes. You will get into trouble using hard-coded slashes on regular filesystem paths but not require. What seems to be the issue is that I have not compiled the project using grunt. When I tried that, I was informed I don't have the proper dependencies:

C:\Users\wzhjtn\AppData\Roaming\npm\node_modules\blaze_compiler>grunt compile
>> Local Npm module "grunt-contrib-nodeunit" not found. Is it installed?
>> Local Npm module "grunt-typescript" not found. Is it installed?
Warning: Task "typescript:base" not found. Use --force to continue.

Aborted due to warnings.

From the documentation, I wasn't aware I had to compile the project from source to use it. Let me do that and get back to you.

**Comment by ShamariFeaster** [comment]
Solution: I had to install "grunt-contrib-nodeunit" and "grunt-typescript" in the project's local "node_modules" directory then run "grunt compile" which generated the necessary ".js" files. 

After that I was able to successfully compile yaml to a rules file. thanks for your responses. I don't know  enough about your project to say there's an error on your end but it appears you are expecting the repo that users pull down from nmp to already have the compiled .js files, which it currently does not.
